tests/framework/namespace.go

```diff
@@ -0,0 +1,78 @@
+package framework
+
+import (
+	"fmt"
+	"strings"
+	"time"
+
+	. "github.com/onsi/gomega"
+
+	k8sv1 "k8s.io/api/core/v1"
+	"k8s.io/apimachinery/pkg/api/errors"
+	metav1 "k8s.io/apimachinery/pkg/apis/meta/v1"
+	"kubevirt.io/kubevirt/pkg/kubecli"
+	ktests "kubevirt.io/kubevirt/tests"
+)
+
+func CreateNamespaces() {
+	virtCli, err := kubecli.GetKubevirtClient()
+	ktests.PanicOnError(err)
+
+	testNamespaces := []string{ktests.NamespaceTestDefault, ktests.NamespaceTestAlternative}
+	// Create a Test Namespaces
+	for _, namespace := range testNamespaces {
+		ns := &k8sv1.Namespace{
+			ObjectMeta: metav1.ObjectMeta{
+				Name: namespace,
+			},
+		}
+		_, err = virtCli.CoreV1().Namespaces().Create(ns)
+		if !errors.IsAlreadyExists(err) {
+			ktests.PanicOnError(err)
+		}
+	}
+}
+
+func RemoveNamespaces() {
+	virtCli, err := kubecli.GetKubevirtClient()
+	ktests.PanicOnError(err)
+	testNamespaces := []string{ktests.NamespaceTestDefault, ktests.NamespaceTestAlternative}
+
+	// First send an initial delete to every namespace
+	for _, namespace := range testNamespaces {
+		err := virtCli.CoreV1().Namespaces().Delete(namespace, nil)
+		if !errors.IsNotFound(err) {
+			ktests.PanicOnError(err)
```

Comment:
the usage of PanicOnError is problematic, the execution will die and you end up with nothing. this is especially wrong to do when the execution is about to finish, we will lose all tests results

Reply:
could you give me an example how do we report errors ?